dataikuapi/dss_plugin_mlflow/utils.py

```diff
@@ -62,30 +62,39 @@ def __init__(self, client, project, managed_folder, host=None):
                 "DSS_MLFLOW_INTERNAL_TICKET": self.client.internal_ticket
             })
 
-        if not isinstance(managed_folder, DSSManagedFolder):
-            raise TypeError('managed_folder must a DSSManagedFolder.')
-
         if not client._session.verify:
             self.mlflow_env.update({"MLFLOW_TRACKING_INSECURE_TLS": "true"})
         elif isinstance(client._session.verify, str):
             self.mlflow_env.update({"MLFLOW_TRACKING_SERVER_CERT_PATH": client._session.verify})
 
+        mf_full_id = managed_folder
+        if isinstance(managed_folder, DSSManagedFolder):
+            mf_full_id = managed_folder.project.project_key + "." + managed_folder.id
+        elif hasattr(managed_folder, 'name'): # True if dataiku.Folder
+            mf_full_id = managed_folder.name
+
+        if not isinstance(mf_full_id, str):
+            raise TypeError('Type of managed_folder must be "str", "DSSManagedFolder" or "dataiku.Folder".')
```

Comment:
I would have gone for:
```
if isinstance(managed_folder, DSSManagedFolder):
    mf_full_id = managed_folder.project.project_key + "." + managed_folder.id
elif isinstance(managed_folder, str):
    mf_full_id = managed_folder.name
else:
    try:
        from dataiku import Folder
        if isinstance(managed_folder, Folder):
            mf_full_id = managed_folder.name
    except ImportError:
        pass

if not mf_full_id:
   raise TypeError('Type of managed_folder must be "str", "DSSManagedFolder" or "dataiku.Folder".')
```

Reply:
done and retested by running locally my tests in integration/python/scenario/ml/prediction/mlflow/experiment_tracking/test_client_plugin.py